### ISSUE TITLE
Convert BL+FEL+RPU to BL+MEL+RPU or BL+RPU

### ISSUE THREAD

**Issue by nohajc** [quietvoid/dovi_tool#10]
Hi, I'd like to "remux" profile 7 FEL mkv to profile 7 MEL mkv or to profile 8 mkv. Is it possible to achieve this?

I know this tool lets me demux BL and EL+RPU into two separate hevc files or extract RPU as a separate file. However, I am not sure what to do with those files next. I am aware the dolby mp4muxer can be used to combine BL and EL files and mux them to mp4 but I'd like to avoid this step if possible.

Basically, my goal is to produce another mkv which will have the original FEL converted to MEL or dropped completely.

**Comment by quietvoid** [comment]
There is no support for converting (or discarding) the layer without demuxing, currently.

There was an experiment in the `convert_rpu` branch, but the resulting files did not playback properly.
PRs welcome to fix it, as I don't have time to work on this in my free time.

**Comment by nohajc** [comment]
Thanks for the quick response!

Let's say I just want to convert FEL to MEL. I can use your dovi_tool in mode 1 which will produce BL and EL files. Now I need to combine these into a single track again and mux into mkv.

Are you saying MakeMKV can do that? (i.e. take raw hevc input as opposed to full BD image)

Or do I create mkv with two video tracks first and then somehow merge them?

I suppose I could also use mp4muxer and remux that mp4 back to mkv but the dolby's tool seems kind of slow.

**Comment by quietvoid** [comment]
> Or do I create mkv with two video tracks first and then somehow merge them?
Yes, MakeMKV merges them back I think.

**Comment by nohajc** [comment]
Yeah, I've seen references to that script all over MakeMKV forums but it looks like there's no repository. I could only find a bunch of mega.nz links scattered through various posts. :D

Anyway, I think this can be closed. It was more of a question than a feature request. Thanks again.

**Comment by quietvoid** [comment]
Eventually my goal for this tool is to have these features:

- Muxing BL, EL together
- Convert RPU, or discard EL (keeping RPU) and output as single layer
- Injecting RPU in between frames

No idea if and when any of those will be done though.
